src/overlays/fbdemos/ovl_fbdemo_triforce/z_fbdemo_triforce.c

```diff
@@ -15,35 +16,132 @@ void TransitionTriforce_SetType(void* thisx, s32 type);
+            1000.0f, 1.0f);
+    this->transPos = 1.0f;
+    this->state = STATE_SPIRAL_IN_FAST;
+    this->step = 0.015f;
+    this->fadeType = TYPE_TRANSPARENT_TRIFORCE;
+    return this;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Update.s")
+void TransitionTriforce_Destroy(void* thisx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_SetColor.s")
+void TransitionTriforce_Update(void* thisx, s32 updateRate) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_SetType.s")
+    for (i = updateRate; i > 0; i--) {
+        if (this->state == STATE_SPIRAL_IN_SLOW) {
+            this->transPos = CLAMP_MIN(this->transPos * (1.0f - this->step), 0.03f);
+        } else if (this->state == STATE_SPIRAL_IN_FAST) {
+            this->transPos = CLAMP_MIN(this->transPos - this->step, 0.03f);
+        } else if (this->state == STATE_SPIRAL_OUT_SLOW) {
+            this->transPos = CLAMP_MAX(this->transPos / (1.0f - this->step), 1.0f);
+        } else if (this->state == STATE_SPIRAL_OUT_FAST) {
+            this->transPos = CLAMP_MAX(this->transPos + this->step, 1.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/func_80AC5280.s")
+void TransitionTriforce_SetColor(void* thisx, u32 color) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_Draw.s")
+    this->color.rgba = color;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_triforce/TransitionTriforce_IsDone.s")
+void TransitionTriforce_SetType(void* thisx, s32 type) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+
+    this->fadeType = type;
+}
+
+void TransitionTriforce_SetState(void* thisx, s32 state) {
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+
+    this->state = state;
+}
+
+void TransitionTriforce_Draw(void* thisx, Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+    Mtx* modelView;
+    s32 pad[2];
+    TransitionTriforce* this = (TransitionTriforce*)thisx;
+    f32 rotation = this->transPos * 360.0f;
+
+    modelView = this->modelView[this->frame];
+    this->frame ^= 1;
+    guScale(&modelView[0], this->transPos * 0.625f, this->transPos * 0.625f, 1.0f);
+    guRotate(&modelView[1], rotation, 0.0f, 0.0f, 1.0f);
+    guTranslate(&modelView[2], 0.0f, 0.0f, 0.0f);
+    gDPPipeSync(gfx++);
+    gSPDisplayList(gfx++, sTriforceWipeDL);
+    gDPSetColor(gfx++, G_SETPRIMCOLOR, this->color.rgba);
+    gDPSetCombineMode(gfx++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+    gSPMatrix(gfx++, &this->projection, G_MTX_LOAD | G_MTX_PROJECTION);
+    gSPMatrix(gfx++, &modelView[0], G_MTX_LOAD);
+    gSPMatrix(gfx++, &modelView[1], G_MTX_NOPUSH | G_MTX_MODELVIEW);
+    gSPMatrix(gfx++, &modelView[2], G_MTX_NOPUSH | G_MTX_MODELVIEW);
```

Comment:
Always write 3 flags in `gSPMatrix`
```suggestion
    gSPMatrix(gfx++, &this->projection, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_PROJECTION);
    gSPMatrix(gfx++, &modelView[0], G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
    gSPMatrix(gfx++, &modelView[1], G_MTX_NOPUSH | G_MTX_MUL | G_MTX_MODELVIEW);
    gSPMatrix(gfx++, &modelView[2], G_MTX_NOPUSH | G_MTX_MUL | G_MTX_MODELVIEW);
```